src/lib/image_hash/average_hash.h

```diff
@@ -37,72 +37,39 @@ namespace image_hash
 		return res;
 	}
 
-	inline ahash_result special_case_fuzzy_ahash(const cv::Mat& gray, unsigned mode)
+	template <unsigned CELLSIZE>
+	inline ahash_result<CELLSIZE> fuzzy_ahash(const cv::Mat& img, uchar threshold=0, unsigned mode=ahash_result<CELLSIZE>::ALL)
```

Comment:
Removed some dead code.

This, sadly, now takes a template param. Could you tell?